### ISSUE TITLE
F821 UndefinedName false positive on annotated global

### ISSUE THREAD

**Issue by andersk** [charliermarsh/ruff#1194]
As of 26d3ff5a3a235cda4883f62e4cbce4096c5c9388 (#1166), Ruff gives an `F821` error here that Flake8 does not. Deleting the `n: int` annotation makes the error disappear.

```python
n: int


def f():
    print(n)


def g():
    global n
    n = 1


g()
f()
```

```console
$ flake8 test.py
$ ruff test.py
Found 1 error(s).
test.py:5:11: F821 Undefined name `n`
```

**Comment by charliermarsh** [comment]
Will fix today!
